### ISSUE TITLE
Piston activating sometimes causes inventory items and text to glitch out

### ISSUE THREAD

**Issue by Karotte128** [IrisShaders/Iris#2532]
### What happened?

Activating a sticky piston with redstone sometimes causes the items in the hotbar and inventory to turn black. Also all text (chat, buttons, ...) turns black. Reloading shader packs does not work, issue even exists on title screen after leaving world. Can be temporarily fixed by mining any block.

### Screenshots

![Image](https://github.com/user-attachments/assets/314906c1-bcb8-4da0-a190-3150d2223268)
![Image](https://github.com/user-attachments/assets/bed2d787-a700-400a-a06d-d1d0bdb08b5a)
![Image](https://github.com/user-attachments/assets/512bbd74-bef5-4e42-821e-0525c8629d3f)

### Log output

none

### Minecraft Version

1.21.3

### Iris Version

1.8.0+mc1.21.3

### Operating System

macOS 15.1

### What is your GPU?

Apple M1 Ultra

### Additional context

Tested with multiple shader packs (Complementary, BSL, ...) Only happens on 1.21.3 Fabric and NeoForge.
Other mods: Sodium, Fabric API

**Comment by exsesx** [comment]
Thanks for highlighting this! I’ve had the same shader rendering issue with Iris and thought it was just my setup. Glad to see it’s not just me—looking forward to a fix!

**Comment by kleinesHu** [comment]
same issue here (M3 Max, Fabric, Iris, Sodium) - it always happens when holding no item in my hand and activating piston - items restore when changing to filled slot in inventory

**Comment by Karotte128** [comment]
Also happens on 1.21.4

**Comment by cartography-is-cool** [comment]
Yep, same here, using an M1 MBP. As soon as any shaders are loaded (Complementary or Photon) and a piston activates, the GUI & other faults described above happen. Always get the following line in the log when it happens:

UNSUPPORTED (log once): POSSIBLE ISSUE: unit 0 GLD_TEXTURE_INDEX_2D is unloadable and bound to sampler type (Float) - using zero texture because texture unloadable

**Comment by keith-ferney** [comment]
Same issue here 1.21.3 on a M2 Pro

**Comment by Karotte128** [comment]
Is this issue solved with the latest release?

**Comment by hylasrc** [comment]
Same here.
It seems to have not been fixed.
